### PR TITLE
feat: Add trackerbot

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ebr_trackerbot/checker.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+def check_tests(
+    loop, check_tests_delay, storage, slack_client, api_url, br_url
+):  # pylint: disable-msg=too-many-arguments
+    """
+    Check tests and send slack message to user when some tracked tests failed
+    """
+    logging.info("Check tests results")
+    loop.call_later(check_tests_delay, check_tests, loop, check_tests_delay, storage, slack_client, api_url, br_url)
+    end = pendulum.now("Europe/Berlin")
```
Comment (eugene-davis) [inline ebr_trackerbot/checker.py]: Lets default to UTC, and maybe allow the client to specify timezone in their request

Comment (eugene-davis) [reply ebr_trackerbot/checker.py]: Alternatively, check to see if there is a config option for the timezone, and if not default to UTC

File: ebr_trackerbot/storage/sqlite.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+"""
+Slack Bot SQLite Storage
```
Comment (eugene-davis) [inline ebr_trackerbot/storage/sqlite.py:2]: We need an issue tracking the idea to move this to a generic DB connector in the future

File: ebr_trackerbot/bot.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+
+def main():
+    """
+    Initialize slack bot
+    """
+    logging.basicConfig(
+        level=logging.DEBUG, format="[%(asctime)s] [%(levelname)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+    )
+    if "API_URL" not in os.environ:
```
Comment (eugene-davis) [inline ebr_trackerbot/bot.py:119]: We need to create an issue about migrating configuration into vault-anyconfig

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM python:3
```
Comment (eugene-davis) [inline Dockerfile]: Why not use the standard dockerfile?

**Review by eugene-davis — CHANGES_REQUESTED** [review]

File: ebr_trackerbot/bot.py (hunk 3 above)
Comment (eugene-davis) [reply ebr_trackerbot/bot.py:119]: Captured in #2

File: ebr_trackerbot/storage/sqlite.py (hunk 2 above)
Comment (eugene-davis) [reply ebr_trackerbot/storage/sqlite.py:2]: Captured in #3

**Comment by eugene-davis** [comment]
Also, the README needs to be updated. Can you include (at least) in it:
1. Ports required
2. Environmental variables configured

Also I have a feeling that the README from the template didn't get included, maybe you can start from that?

File: ebr_trackerbot/bot.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    Listener for slack message event
+    """
+    data = payload["data"]
+    try:
+        channel_id = data["channel"]
+        thread_ts = data["ts"]
+        user = data["user"]
+        msg_id = data["client_msg_id"]
+    except KeyError:
+        logging.debug("Missing one of channel, ts, user, client_msg_id in slack message")
```
Comment (eugene-davis) [inline ebr_trackerbot/bot.py]: Small change, add a colon, i.e. "Missing one of: channel, ts, user ...

File: ebr_trackerbot/bot.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+def get_storage():
+    """
+    Returns slack bot storage
+    """
+    if not hasattr(STATE, "bot_storage"):
+        STATE.bot_storage = []
+    return list(filter(lambda x: x["name"] == get_storage_name(), STATE.bot_storage))[0]
+
+
+def say_hello(**payload):
```
Comment (eugene-davis) [inline ebr_trackerbot/bot.py]: Can we make this a better name, that is more reflective of what it does?

File: Dockerfile (hunk 4 above)
Comment (ghost) [reply Dockerfile]: ok, I migrated to standard dockerfile

File: ebr_trackerbot/bot.py (hunk 5 above)
Comment (ghost) [reply ebr_trackerbot/bot.py]: fixed

File: ebr_trackerbot/checker.py (hunk 1 above)
Comment (ghost) [reply ebr_trackerbot/checker.py]: I changed the default to UTC

File: ebr_trackerbot/bot.py (hunk 6 above)
Comment (ghost) [reply ebr_trackerbot/bot.py]: ok

File: README.md [hunk 7]
```diff
@@ -1,13 +1,47 @@
...
+## Features
+
+
+**Tracker bot use configuration provided via environment variables**
+
+* `SLACK_TOKEN` - (required) slack token which is used to connect slack server
+* `API_URL` - (required) url to br board api endpoint. Used for periodical checks for failed tests.
+* `BACKEND` - (optional) backend storage specification (memory or sqlite). Default is memory.
+* `SQLITE_FILENAME` - (optional) sqlite filename path. Default is data.db
+* `BR_URL` - (optional) link to br board which is used in slack message when test failed
```
Comment (eugene-davis) [inline README.md]: Except this is about to become more generic right? :)

File: README.md [hunk 8]
```diff
@@ -1,13 +1,47 @@
...
+
+
+**How to build:**
```
Comment (eugene-davis) [inline README.md]: Also, building the actual python package. Though both it and the docker build are general enough that they could probably be left out safely.

File: README.md [hunk 9]
```diff
@@ -1,13 +1,47 @@
...
 
 `
 docker build . -t test-tracker-slack-bot
 `
 
+
+**How to run:**
+
 `
 docker run -e BR_URL=<br board url> -e API_URL=<api url> -e SLACK_TOKEN=<token> -e BACKEND=<memory|sqlite> -e SQLITE_FILENAME=<filename> test-tracker-slack-bot python ebr-trackerbot
```
Comment (eugene-davis) [inline README.md]: Can we reference the image that will be pushed to Bintray here? It should be `tomtom-docker-registry.bintray.io/python/ebr_trackerbot` if I'm not mistaken.

**Review by eugene-davis — APPROVED** [review]

**Pull request merged**